Library/Homebrew/language/node.rb

```diff
@@ -4,6 +4,17 @@ def self.npm_cache_config
       "cache=#{HOMEBREW_CACHE}/npm_cache\n"
     end
 
+    def self.pack_for_installation
+      # npm 5.0.0 broke the assumption Homebrew made that npm would always
+      # create and install real directories when passed a file path. We can
+      # restore the old behaviour by packing the path first with "npm pack"
+      # which also helpfully spits out the filename it just created allowing
+      # us to pass that as the final argument.
```

Comment:
Hmm. I don't know how to explain it with brevity really, it's a fairly big change.

Essentially when you used to `npm install some-directory --global --prefix="blah"` you could then dispose of your build directory & purely retain the end product, the "blah" prefix in this example. From the new `npm` if you pass it a directory to install it only installs symlinks back to that directory in the "blah" prefix, which doesn't work for Homebrew since we trash `buildpath`s/`testpath`s the second we're done with 'em normally.

`npm pack` is supported as an upstream way of retaining the old behaviour Homebrew relies on to be able to sensibly package `npm`-using formulae.